packages/block-editor/src/components/block-list/index.js

```diff
@@ -46,16 +51,47 @@ function BlockList(
 			isDraggingBlocks,
 		} = select( 'core/block-editor' );
 
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const multiSelectedBlockClientIds = getMultiSelectedBlockClientIds();
+
+		// Determine if there is an active template part (or other entity) to spotlight.
+		// Restrict to Full Site Editing experiment.
+		const isFullSiteEditingEnabled = getSettings()
```

Comment:
Why are we restricting to FSE here? I can see a world where this is useful for reusable blocks if they are refactored to work like template parts.

Reply:
Mostly out of caution for the time being as the only block currently being tested in this is the template part.  But I agree, this could definitely be useful for non-FSE things.  Its just that the designs that have driven this to life have only thought about template parts for now.

Reply:
Now it is just there because the only block in the list is of allowed blocks is only registered in FSE, so it stops the selector from running unnecessarily.  We could remove this if desired, or if other blocks are added to this functionality.

Reply:
I think the selector still has to do the same "expensive" calculations to figure out if a block is not activee

Reply:
> I think the selector still has to do the same "expensive" calculations to figure out if a block is not active

Ah gotcha.  It should never change either, so maybe we run it once to get the value on mount or something?  Or... just get rid of it?

Reply:
Or maybe it would make sense to make an `__experimentalIsFullSiteEditingActive` selector that is memoized? 🤔

Reply:
> I think the selector still has to do the same "expensive" calculations to figure out if a block is not active

Im not sure I follow.  `getSettings` ? Or I guess the new selector since you are mentioning the block being active? Luckily we aren't calling it on/for every block, just on the block list and only for the selected block. 🤔

Reply:
> Now it is just there because the only block in the list is of allowed blocks is only registered in FSE, so it stops the selector from running unnecessarily

I actually misunderstood what you were saying! I read this as "the list of allowed blocks stops the selector from running unnecessarily because the allowed blocks are only registered when FSE is active". So you can disregard my comment 😅